### DISCUSSION TITLE
release version crash on m1 or m2, os version is 14.*

### DISCUSSION THREAD

**Discussion by dttlgotv** [ExistentialAudio/BlackHole#765] [Q&A]
Anyone meet this issue? I use tag v0.5.0 to compile release version well. After call "sudo launchctl kickstart -kp system/com.apple.audio.coreaudiod" on M1 or M2 machine ，os is 14.*, the system micphone and speakers will lost. And midi is hung up. It seems that core audio is crash. From system log, it can be seen:
Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0xffffffffffffffb8
Exception Codes:       0x0000000000000001, 0xffffffffffffffb8

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [3112]

VM Region Info: 0xffffffffffffffb8 is not in any region.  Bytes after previous region: 18446743592673214393  
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      commpage (reserved)        1000000000-7000000000   [384.0G] ---/--- SM=NUL  ...(unallocated)
--->  
      UNUSED SPACE AT END

Thread 0::  Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	       0x18de788b4 mach_msg2_trap + 8
1   libsystem_kernel.dylib        	       0x18de8ad30 mach_msg2_internal + 80
2   libsystem_kernel.dylib        	       0x18de814f0 mach_msg_overwrite + 476
3   libsystem_kernel.dylib        	       0x18de78c38 mach_msg + 24
4   CoreAudio                     	       0x1908a3718 HALC_ProxyObject::GetPropertyData(AudioObjectPropertyAddress const&, unsigned int, void const*, unsigned int, unsigned int&, void*) const + 4144
5   CoreAudio                     	       0x1907d0808 HALC_ShellObject::GetPropertyData(unsigned int, AudioObjectPropertyAddress const&, unsigned int, void const*, unsigned int, unsigned int&, void*) const + 1244
6   CoreAudio                     	       0x19069decc HAL_HardwarePlugIn_ObjectGetPropertyData(AudioHardwarePlugInInterface**, unsigned int, AudioObjectPropertyAddress const*, unsigned int, void const*, unsigned int*, void*) + 556
7   CoreAudio                     	       0x1907b3974 HALPlugIn::ObjectGetPropertyData(HALObject const&, AudioObjectPropertyAddress const&, unsigned int, void const*, unsigned int&, void*) const + 88
8   CoreAudio                     	       0x1904f9f8c HALStream::IsInput() const + 116
9   CoreAudio                     	       0x1904f9d54 HALDevice::ObjectPublished(HALObject*, CAPropertyAddressList&) + 108
10  CoreAudio                     	       0x190755c0c HALObject::ObjectsPublishedAndDied(unsigned int, unsigned int const*, unsigned int, unsigned int const*) + 472
11  CoreAudio                     	       0x1905cd83c HALSystem::AudioObjectsPublishedAndDied(AudioHardwarePlugInInterface**, unsigned int, unsigned int, unsigned int const*, unsigned int, unsigned int const*) + 208
12  CoreAudio                     	       0x190597444 HALC_ShellDevice::RebuildStreamLists(bool) + 1788
13  CoreAudio                     	       0x1905a038c HALC_ShellDevice::Activate() + 68
14  CoreAudio                     	       0x190609854 HALC_ShellObjectMap::CreateObject(unsigned int, unsigned int, unsigned int, unsigned int) + 1300
15  CoreAudio                     	       0x19069bb3c HALC_ShellPlugIn::_ReconcileDeviceList(bool, bool, std::__1::vector<unsigned int, std::__1::allocator<unsigned int>>&, std::__1::vector<unsigned int, std::__1::allocator<unsigned int>>&) + 992
16  CoreAudio                     	       0x1909d53f8 HALB_CommandGate::ExecuteCommand(void () block_pointer) const + 232
17  CoreAudio                     	       0x1907d022c HALC_ShellObject::ExecuteCommand(void () block_pointer) const + 80
18  CoreAudio                     	       0x19069b630 HALC_ShellPlugIn::ReconcileDeviceList(bool, bool) + 236
19  CoreAudio                     	       0x19069cc00 HALC_ShellPlugIn::ConnectToServer() + 212
20  CoreAudio                     	       0x19069ec6c HAL_HardwarePlugIn_InitializeWithObjectID(AudioHardwarePlugInInterface**, unsigned int) + 364
21  CoreAudio                     	       0x1904f782c HALPlugInManagement::CreateHALPlugIn(HALCFPlugIn const*) + 1728
22  CoreAudio                     	       0x1904f6a2c HALSystem::InitializeDevices() + 2468
23  CoreAudio                     	       0x1904f3c8c HALSystem::CheckOutInstance() + 232
24  CoreAudio                     	       0x1907601e8 AudioObjectAddPropertyListener_mac_imp + 92
25  coreaudiod                    	       0x1048fb048 main + 400
26  dyld                          	       0x18db3d0e0 start + 2360

After I use debug version to test, everything is well. Another in other os version or intel (os is 14.*) everything is well too.

**Comment by devinroth** [comment]
So the debug build doesn’t cause a crash but release does? That seems like a compiler issue.

**Comment by devinroth** [comment]
What you can do to debug is find the difference between the debug and release and slowly change the options until you find the crash.
